packages/astro/src/vite-plugin-astro-server/route.ts

```diff
@@ -381,7 +381,8 @@ async function getScriptsAndStyles({ pipeline, filePath }: GetScriptsAndStylesPa
 			children: '',
 		});
 
-		if (settings.config.experimental.devOverlay) {
+		const devOverlayEnabled = await settings.preferences.get('devOverlay.enabled');
+		if (settings.config.experimental.devOverlay && devOverlayEnabled) {
```

Comment:
Question: If there's a scenario where multiple people are working on a project, a majority prefers to not have the overlay, but I prefer to have it, this condition makes it a little hard to work for me. Since I'd have to enable `experimental.devOverlay` still and inform everyone else to disable it in their preference.

I think I'd expect the inverse, where I can enable it in my preference so no changes are needed from everyone else. In which case, I think a condition like this instead:

```js
if (devOverlayEnabled ?? settings.config.experimental.devOverlay)
```

Where `devOverlayEnabled` is `boolean | undefined`, and `undefined` means unset, would make this work 🤔

Reply:
This feature makes more sense when you realize that the devOverlay will be enabled by default once we remove the experimental flag (happening in another PR)! I believe the preference will be the only way to turn it off, but I could be wrong about that.

Reply:
In that case, I think we can revisit the `boolean | undefined` approach then for future options 👍  If there will be a request for an explicit config option to disable the dev overlay in the future, we might have to tweak this to work differently.

Reply:
Yeah, me and Nate had a discussion about it yesterday, for this first version we're making it so the config always win (the idea is that if your project doesn't support the overlay for some reason, you can disable it for everyone). 

I don't necessarily have particular thoughts on what the behaviour should be, to be honest, open to whatever we deem is best.

Reply:
Correct, the resolution for now is that the config setting wins out and preferences can only _disable_ a feature. If we want to revisit that in the future, we definitely can.